impl/src/main/java/org/ehcache/config/serializer/DefaultSerializationProviderConfiguration.java

```diff
@@ -41,40 +42,42 @@
     if (serializerClass == null) {
       throw new NullPointerException("Serializer class cannot be null");
     }
-    String alias = serializableClass.getName();
     
-    if (serializerClass.isAnnotationPresent(Serializer.Transient.class)) {
-      if (serializerClass.isAnnotationPresent(Serializer.Persistent.class)) {
-        if (transientSerializers.containsKey(alias)) {
-          if (persistentSerializers.containsKey(alias)) {
-            throw new IllegalArgumentException("Duplicate transient & persistent serializer for class : " + alias);
-          } else {
-            persistentSerializers.put(alias, serializerClass);
-          }
-        } else if (persistentSerializers.containsKey(alias)) {
-          transientSerializers.put(alias, serializerClass);
-        } else {
-          persistentSerializers.put(alias, serializerClass);
-          transientSerializers.put(alias, serializerClass);
-        }
+    String alias = serializableClass.getName(); 
```

Comment:
I understand that this new code is logically equivalent to what was there before, however that isn't entirely clear when looking at the changeset.  The only thing that needs to change to accomplish the task is to change the conditions in the if statements to `isConstructorPresent(...)`.